### PR TITLE
Allow to_pandas to return pandas.ArrowDtype

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Adds a `arrow_type: bool` parameter to `to_pandas` to allow the conversion to return `pandas.ArrowDtype` in pandas.

(Opens up the dream of cudf to pandas round tripping to happen via arrow formatted data)

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_series.py [hunk 1]
```diff
@@ -2708,3 +2708,44 @@ def test_series_from_large_string():
...
+        datetime.timedelta(1),
+        {"1": 2},
+        [1],
+        decimal.Decimal("1.0"),
+    ],
+)
+def test_series_to_pandas_arrow_type_nullable_raises(scalar):
+    pa_array = pa.array([scalar, None])
+    ser = cudf.Series(pa_array)
+    with pytest.raises(ValueError):
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_series.py:2729]: For readability, perhaps specify the reason argument `reason="cannot both be set"`?

**Review by isVoid — APPROVED** [review]

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -2043,12 +2052,22 @@ def to_pandas(
         2     NaN
         3    30.0
         dtype: float64
+
+        ``arrow_type=True`` converts the result to ``pandas.ArrowDtype``:
```
Comment (galipremsagar) [inline python/cudf/cudf/core/series.py:2056]:
```suggestion
        ``arrow_type=True`` converts the result to ``pandas.ArrowDtype``:

```

For this to render correctly we might need a new-line

**Review by galipremsagar — APPROVED** [review]
LGTM! A minor comment..

**Pull request merged**